addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -0,0 +1,292 @@
+                            ('cancel', 'Cancelled')], string='Status', track_visibility='onchange', required=True, copy=False, default='draft')
+    multiple_rfq_per_supplier = fields.Boolean(string='Multiple RFQ per vendor')
+    account_analytic_id = fields.Many2one('account.analytic.account', string='Analytic Account')
+    picking_type_id = fields.Many2one('stock.picking.type', string='Picking Type', required=True, default=_default_get_picking_in)
+
+    @api.multi
+    def tender_cancel(self):
+        # try to set all associated quotations to cancel state
+        for tender in self:
+            for purchase_order in tender.purchase_ids:
+                purchase_order.button_cancel()
+                purchase_order.message_post(body=_('Cancelled by the tender associated to this quotation.'))
+        return self.write({'state': 'cancel'})
+
+    @api.multi
+    def tender_in_progress(self):
+        if not all(obj.line_ids for obj in self):
+            raise UserError(_('You can not confirm call because there is no product line.'))
+        return self.write({'state': 'in_progress'})
+
+    @api.multi
+    def tender_open(self):
+        return self.write({'state': 'open'})
+
+    @api.multi
+    def tender_reset(self):
+        self.write({'state': 'draft'})
+        for p_id in self:
+            # Deleting the existing instance of workflow for PO
+            p_id.delete_workflow()
+            p_id.create_workflow()
+        return True
+
+    @api.multi
+    def tender_done(self):
+        return self.write({'state': 'done'})
+
+    @api.multi
+    def open_product_line(self):
+        """ This opens product line view to view all lines from the different quotations, groupby default by product and partner to show comparaison
+            between vendor price
+            @return: the product line tree view
+        """
+        res = self.env['ir.actions.act_window'].for_xml_id('purchase_requisition', 'purchase_line_tree')
+        res['context'] = self.env.context
+        res['context'] = {
+            'search_default_groupby_product': True,
+            'search_default_hide_cancelled': True,
+            'tender_id': self[0].id,
+        }
+        res['domain'] = [('id', 'in', self[0].po_line_ids.ids)]
+        return res
+
+    @api.multi
+    def open_rfq(self):
+        """ This opens rfq view to view all quotations associated to the call for tenders
+            @return: the RFQ tree view
+        """
+        res = self.env.ref('purchase.purchase_rfq')
+        res['context'] = self.env.context
+        res['domain'] = [('id', 'in', self[0].purchase_ids.ids)]
+        return res
+
+    def _prepare_purchase_order(self, supplier):
+        return {
+            'origin': self.name,
+            'date_order': self.date_end or fields.datetime.now(),
+            'partner_id': supplier.id,
+            'currency_id': self.company_id and self.company_id.currency_id.id,
```

Comment:
No need to check the condition. Accessing a relational field always returns a recordset, empty if the field is not set.